src/templates/finops-hub/modules/storage.bicep

```diff
@@ -97,12 +100,18 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' = {
   }
   kind: 'BlockBlobStorage'
   tags: union(tags, contains(tagsByResource, 'Microsoft.Storage/storageAccounts') ? tagsByResource['Microsoft.Storage/storageAccounts'] : {})
-  properties: {
+  properties: union(!enableInfrastructureEncryption ? {} : {
+    encryption: {
+      keySource: 'Microsoft.Storage'
+      requireInfrastructureEncryption: enableInfrastructureEncryption
+    }
+  }, {
     supportsHttpsTrafficOnly: true
     allowSharedKeyAccess: true
     isHnsEnabled: true
     minimumTlsVersion: 'TLS1_2'
     allowBlobPublicAccess: false
+  })
```

Comment:
This needs to be moved down a few lines. I think you need to update this branch, tho. These changes already went in with the other PR.